### PR TITLE
DraftService: enforce lot's one_service_limit using database constraint

### PR DESCRIPTION
**Pull request by risicle**
~**I'm not planning on merging this PR directly, at merge time I will take the first two commits and submit them separately.** This is just here to allow people to review it conveniently.~ This _is_ now the second half., the first half having been merged as #1005 

https://trello.com/c/bqmbNLBu

Please read commit comments for more detailed information - I've tried to document it step by step.

In short, it's actually implemented largely in the abstract `ServiceTableMixin` and only _enabled_ when the concrete model contains a `lot_one_service_limit` field. For now, that's only `DraftService`.

It would be good to test the first two commits work fine for you on their own before testing the full branch...

### CODE REVIEW

File: app/models/main.py [hunk 1]
```diff
@@ -1266,6 +1274,7 @@ def copy(self):
             supplier=self.supplier,
             data=data,
             status='not-submitted',
+            lot_one_service_limit=self.lot.one_service_limit,
```
Comment (katstevens) [inline app/models/main.py]: This covers both copying from a live Service, and copying from another DraftService, correct?

Comment (risicle) [reply app/models/main.py]: I think so, yes.

Comment (risicle) [reply app/models/main.py]: Well, it's `from_service` that does the former.

**Comment by katstevens** [comment]
First two commits work fine for me locally (unit tests, draft-service related functional tests for G11 and DOS4, some manual testing) 👍

**Comment by risicle** [comment]
Yippee thanks. Weirdly travis is failing but only for python 3.7. Not 3.6, not 3.8.

**Comment by benvand** [comment]
Does this rely on the current duplication being sorted out?

**Comment by risicle** [comment]
> Does this rely on the current duplication being sorted out?

YES.

It will be unhappy if applied to a non-compliant database.

**Comment by katstevens** [comment]
We can use the `delete_draft_service` API method to address the inconsistency on the DraftService table.
